WikiWatershed/model-my-watershed
# Switch to using correctly aligned NLCD19 layers, wipe cached results

## Overview

Copy of #3549 for the hotfix.

Also adds a migration to clear out existing results for GWLF-E projects that used the unaligned 2019 layers since 1.33.0 was released in January 2022. All these projects will have their values recalculated when they are accessed again, allowing us to rebuild the project results over time rather than all at once and overloading the system.

Closes #3571 

## Testing Instructions

* On `hotfix/1.33.8`, go to http://localhost:8000/ and login with an account
  * Make one if you don't have one already
* Select / draw a shape, move to Analyze, move to Model, and select Watershed Multi-Year Model
* Once the calculation completes, create an additional scenario
* Refresh the page
  * [x] Ensure the results load quickly, without being recalculated
* Go to the database console
    ```
    ./scripts/manage.sh db_shell
    ```
* Run this query to check that your project has a true value for `gis_data`:
    ```sql
    SELECT gis_data FROM modeling_project WHERE id = XXX;
    ```
  where XXX is your project's id
* Check out this branch and run migrations
    ```
    ./scripts/manage.sh migrate
    ```
* Run this query to check that `gis_data` is now NULL:
    ```sql
    SELECT gis_data FROM modeling_project WHERE id = XXX;
    ```
  where XXX is your project's id
* Go back to your project in the UI
  * [x] Ensure the results are recalculated, evidenced by the loading spinner
  * [x] Ensure that the calculations finish correctly, with values for all tables
* Refresh the page
  * [x] Ensure the results load quickly, without being recalculated
* Run this query to check that your project has a true value for `gis_data`:
    ```sql
    SELECT gis_data FROM modeling_project WHERE id = XXX;
    ```
  where XXX is your project's id
